src/chrome/chromeDebugAdapter.ts

```diff
@@ -472,6 +481,60 @@ export abstract class ChromeDebugAdapter implements IDebugAdapter {
+            let allStopOnEntryBreakpoints = true;
+
+            // If there is a breakpoint which is not a stopOnEntry breakpoint, we appear as if we hit that one
+            // This is particularly done for cases when we end up with a user breakpoint and a stopOnEntry breakpoint on the same line
+            hitBreakpoints.forEach(bp => {
+                if (!this._stopOnEntryBreakpointIdToRequestedFileName.has(bp)) {
+                    notification.hitBreakpoints = [bp];
+                    allStopOnEntryBreakpoints = false;
+                }
+            });
+
+            // If all the breakpoints on this point are stopOnEntry breakpoints
+            // This will be true in cases where it's a single breakpoint and it's a stopOnEntry breakpoint
+            // This can also be true when we have multiple breakpoints and all of them are stopOnEntry breakpoints, for example in cases like index.js and index.bin.js
+
+            if (allStopOnEntryBreakpoints) {
+                // Get the mapped url of the file where execution is paused
+                const pausedScriptId = notification.callFrames[0].location.scriptId;
+                const pausedScriptUrl = this._scriptsById.get(pausedScriptId).url;
+
+                const mappedUrl = this._pathTransformer.scriptParsed(pausedScriptUrl);
+                let normalizedMappedUrl: string;
+                if (mappedUrl !== undefined) {
+                    normalizedMappedUrl = path.normalize(mappedUrl);
+                }
+
+                // If the file has unbound breakpoints, make sure to resolve them first and then decide to continue or not
+                if (this._pendingBreakpointsByUrl.has(normalizedMappedUrl)) {
+                    this.resolvePendingBreakpoint(this._pendingBreakpointsByUrl.get(normalizedMappedUrl))
+                    .then(() => {
+                        this._pendingBreakpointsByUrl.delete(normalizedMappedUrl);
+                        // If none of the breakpoints resolved in the file were at position (1,1) we should continue
+                        if (!this._userBreakpointOnLine1Col1) {
+                            this.chrome.Debugger.resume()
+                                .catch(e => {
+                                    logger.log("Failed to resume due to exception: " + e.message);
+                                });
+                            return;
+                        } else {
+                            // This means that at least one of the breakpoints resolved was at (1,1) so we don't continue
+                            this._userBreakpointOnLine1Col1 = false;
+                        }
+                    });
+                } else {
+                    // The file has no pending breakpoints to resolve. Just continue in this case.
+                    this.chrome.Debugger.resume()
+                        .catch(e => {
+                            logger.log("Failed to resume due to exception: " + e.message);
```

Comment:
Show we also show a message to the user here?

Reply:
Hmm I am unsure about that one. @roblourens What do you think? I guess if this happens, Chrome will be stuck in paused state.